nbodykit/algorithms/paircount_tpcf/estimators.py

```diff
@@ -103,66 +105,59 @@ def LandySzalayEstimator(pair_counter, data1, data2, randoms1, randoms2, logger=
     # make sure we have the randoms
     assert randoms1 is not None
     comm = data1.comm
-
+    
+    if randoms2 is None: randoms2 = randoms1
+        
+    # and randoms - randoms calculation
+    if logger is not None and comm.rank == 0:
+        logger.info("computing randoms1 - randoms2 pair counts")
+    if not R1R2:
+        R1R2 = pair_counter(first=randoms1, second=randoms2, **kwargs)
+    
     # data1 x data2
     if logger is not None and comm.rank == 0:
         logger.info("computing data1 - data2 pair counts")
-    D1D2 = pair_counter(first=data1, second=data2, **kwargs).pairs
-
-    if randoms2 is None:
-        randoms2 = randoms1
-
-    # determine the sample sizes
-    ND1, NR1 = data1.csize, randoms1.csize
-    ND2 = data2.csize if data2 is not None else ND1
-    NR2 = randoms2.csize
+    D1D2 = pair_counter(first=data1, second=data2, **kwargs)
 
     # do data - randoms correlation
     if logger is not None and comm.rank == 0:
         logger.info("computing data1 - randoms2 pair counts")
-    D1R2 = pair_counter(first=data1, second=randoms2, **kwargs).pairs
+    D1R2 = pair_counter(first=data1, second=randoms2, **kwargs)
 
     if data2 is not None:
         if logger is not None and comm.rank == 0:
             logger.info("computing data2 - randoms1 pair counts")
-        D2R1 = pair_counter(first=data2, second=randoms1, **kwargs).pairs
+        D2R1 = pair_counter(first=data2, second=randoms1, **kwargs)
     else:
         D2R1 = D1R2
-
-    # and randoms - randoms calculation
-    if logger is not None and comm.rank == 0:
-        logger.info("computing randoms1 - randoms2 pair counts")
-    R1R2 = pair_counter(first=randoms1, second=randoms2, **kwargs).pairs
+    
+    fDD, fDR, fRD = R1R2.attrs['weightedpairs']/D1D2.attrs['weightedpairs'], R1R2.attrs['weightedpairs']/D1R2.attrs['weightedpairs'], R1R2.attrs['weightedpairs']/D2R1.attrs['weightedpairs']
```

Comment:
we should break this up into multiple lines, but the calculation looks right to me